llvm-bazel/configure.bzl

```diff
@@ -4,10 +4,113 @@
 
 """Configures an LLVM overlay project."""
 
-load(":overlay_directories.bzl", "overlay_directories")
-
 # Directory of overlay files relative to WORKSPACE
-OVERLAY_DIR = "llvm-project-overlay"
+DEFAULT_OVERLAY_PATH = "llvm-project-overlay"
+
+DEFAULT_TARGETS = [
+    "AArch64",
+    "AMDGPU",
+    "ARM",
+    "BPF",
+    "Hexagon",
+    "Lanai",
+    "NVPTX",
+    "PowerPC",
+    "RISCV",
+    "Sparc",
+    "WebAssembly",
+    "X86",
+]
+
+def _is_absolute(path):
+    """Returns `True` if `path` is an absolute path.
+
+    Args:
+      path: A path (which is a string).
+    Returns:
+      `True` if `path` is an absolute path.
+    """
+    return path.startswith("/") or (len(path) > 2 and path[1] == ":")
+
+def _join_path(a, b):
+    if _is_absolute(b):
+        return b
+    return str(a) + "/" + str(b)
+
+def _overlay_directories(repository_ctx):
```

Comment:
Yeah, it was tricky to have in a separate repo rule because then any LLVM-specific configuration would need to be under a different named repo.

Can always share the starlark *code* here if that becomes useful while keeping a simple repo rule structure?

Reply:
Yeah totally understand what happened here. I think have a separate function that takes a repository_ctx (as you have) is the best thing to do here and we can always break that function out if we have a need.

Reply:
Awesome, and thanks for the review!